src/nvim/option.c

```diff
@@ -4195,6 +4255,21 @@ set_num_option (
 
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting && errmsg == NULL) {
+    char_u buf_old[11], buf_new[11], buf_type[7];
+    vim_snprintf((char *)buf_old, 10, "%ld", old_value);
```

Comment:
`len(str(2**64))` (Python) is 20. So value `30` is definitely not due to stringifying integers, even greatest of them.